### PR TITLE
[camera] Fix preview pause orientation

### PR DESCRIPTION
**Pull request by cassioso**
This PR fixes [`lockedCaptureOrientation` not affecting preview orientation when paused](https://github.com/flutter/flutter/issues/102744).
`pausePreview()` in `camera_controller.dart` is not taking `lockedCaptureOrientation` into consideration, leading to a change in the image orientation if the pause happens when the device is in an orientation different than the locked one.

Notice that `startVideoRecording` consider lockedCaptureOrientation:

```dart
Future<void> startVideoRecording() async {
(...)
    value = value.copyWith(
        isRecordingVideo: true,
        isRecordingPaused: false,
        recordingOrientation: Optional<DeviceOrientation>.fromNullable(
            value.lockedCaptureOrientation ?? value.deviceOrientation));
```

But `pausePreview` is missing that:
```dart
Future<void> pausePreview() async {
(...)
    value = value.copyWith(
        isPreviewPaused: true,
        previewPauseOrientation:
            Optional<DeviceOrientation>.of(value.deviceOrientation));
```

Fixes https://github.com/flutter/flutter/issues/102744

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
Thanks for the submission! Sorry for the delay in triaging this; it missed the usual triage cycle that would have processed it since it was a draft at the time.

**Comment by cassioso** [comment]
No problem, I'm glad to be contributing!

**Comment by cassioso** [comment]
Update#2 moved the reproduction steps in the PR description into an issue.

**Comment by cassioso** [comment]
Up?

Also maybe we can have the "needs tests" label removed from this PR.

File: packages/camera/camera/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ## 0.9.4+22
 
+* Fixes preview orientation when pausing preview with locked orientation.
```
Comment (stuartmorgan) [inline packages/camera/camera/CHANGELOG.md:3]: You lost the version change in a merge; that will need to be fixed before CI will pass.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Comment by stuartmorgan** [comment]
Adding @bparrishMines for secondary review.

File: packages/camera/camera/CHANGELOG.md (hunk 1 above)
Comment (cassioso) [reply packages/camera/camera/CHANGELOG.md:3]: Thanks for noting, this is now fixed.

File: packages/camera/camera/test/camera_test.dart [hunk 2]
```diff
@@ -1180,6 +1181,31 @@ void main() {
...
+              name: 'cam',
+              lensDirection: CameraLensDirection.back,
+              sensorOrientation: 90),
+          ResolutionPreset.max);
+      await cameraController.initialize();
+      cameraController.value = cameraController.value.copyWith(
+          isPreviewPaused: false,
+          deviceOrientation: DeviceOrientation.portraitUp,
+          lockedCaptureOrientation:
+              const Optional<DeviceOrientation>.fromNullable(
```
Comment (bparrishMines) [inline packages/camera/camera/test/camera_test.dart:1198]: nit: Same for this one. Should this be using `.of`.

```suggestion
              const Optional<DeviceOrientation>.of(
```

File: packages/camera/camera/lib/src/camera_controller.dart [hunk 3]
```diff
@@ -359,8 +359,8 @@ class CameraController extends ValueNotifier<CameraValue> {
       await CameraPlatform.instance.pausePreview(_cameraId);
       value = value.copyWith(
           isPreviewPaused: true,
-          previewPauseOrientation:
-              Optional<DeviceOrientation>.of(value.deviceOrientation));
+          previewPauseOrientation: Optional<DeviceOrientation>.fromNullable(
```
[the remark this answers is not in the document]
Comment (cassioso) [reply packages/camera/camera/lib/src/camera_controller.dart:362]: Good point, the same in `lockCaptureOrientation`

**Review by bparrishMines — APPROVED** [review]
LGTM

**Comment by DuM4ll3** [comment]
Thanks for submitting this @cassioso, It will help me a lot! Hope it gets approved soon

**Pull request merged**